libs/pulumi-static-site/src/bucket.ts

```diff
@@ -0,0 +1,160 @@
+import * as aws from '@pulumi/aws'
+import * as pulumi from '@pulumi/pulumi'
+
+export class Bucket extends pulumi.ComponentResource {
+    readonly name: string
+    readonly bucket: aws.s3.Bucket
+
+    constructor(
+        name: string,
+        args: {
+            website?: aws.s3.BucketArgs['website']
+        },
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super('swm:pulumi-static-site:bucket/Bucket', name, {}, opts)
+
+        this.name = name
+
+        this.bucket = new aws.s3.Bucket(
+            name,
+            {
+                website: pulumi.output(args.website).apply((website) => ({
+                    ...(website ?? {}),
+                    indexDocument: website?.indexDocument ?? 'index.html',
+                })),
+            },
+            { parent: this },
+        )
+
+        const accountNumber = aws
+            .getCallerIdentity({ parent: this })
+            .then((current) => current.accountId)
+
+        const policy = pulumi
+            .all([this.bucket.arn, accountNumber])
+            .apply(([bucketArn, accountNumber]) =>
+                aws.iam.getPolicyDocument(
+                    {
+                        version: '2012-10-17',
+                        statements: [
+                            {
+                                sid: 'AllowPublicRead',
+                                effect: 'Allow',
+                                principals: [
+                                    { type: 'AWS', identifiers: ['*'] },
+                                ],
+                                actions: ['s3:GetObject', 's3:ListBucket'],
+                                resources: [`${bucketArn}/*`],
+                            },
+                            {
+                                sid: 'AllowDevelopersToUpload',
+                                effect: 'Allow',
+                                principals: [
+                                    {
+                                        type: 'AWS',
+                                        identifiers: [
+                                            `arn:aws:iam::${accountNumber}:role/Wanews-ADFS-Digital-Developers`,
```

Comment:
If we are going to need internal roles in these modules they probably should be in a private repo, not in an OSS one